apollo-at-link/src/argument.rs

```diff
@@ -0,0 +1,136 @@
+use apollo_compiler::ast::Value;
+use apollo_compiler::schema::{Directive, Name};
+use apollo_compiler::{Node, NodeStr};
+use apollo_federation_error::error::{FederationDirectiveErrorCategory, FederationError};
+use std::ops::Deref;
+
+pub fn directive_optional_enum_argument<'directive>(
+    application: &'directive Node<Directive>,
```

Comment:
Wondering if these validations(??) can be on a struct itself rather than as single functions. I am hoping this would help us get rid of this lifetime.